### PR TITLE
Issue #247: Enabel user to chose startup mode and open a note based o…

### LINKED ISSUE #247: Open a default note on startup
<h3>Description</h3>
Currently when you first open the app, it greets you with the home screen and the list of notes on the side. Ideally, it would instead just open up the default note, which could be the most recently opened note. I would like it to open up to one of the already existing notes instead of the home screen.

<h3>Alternatives:</h3>

On Start up:
- Open up the most recently edited note
- Open up the most recently selected note
- Open up a new note
-or-
- Provide options in the settings as to what to do on startup (like Chrome does)

### PR DESCRIPTION
**Pull request by sensokame**
…n it.

Fixes #247  .

Changes proposed in this pull request:
- Add drop down list in settings to chose startup mode
- Load default startup note depending on the mode

Note:
Custom mode may not have a very appealing visual representation (info_icon on the selected node to set as startup note) so it may need revision more than the others.

### CODE REVIEW

**Comment by abahmed** [comment]
Could you update current screenshots too? @sensokame

**Comment by abahmed** [comment]
Test failure, fix it and check with `npm test`
 FAIL  app/components/settings/test.jsx
  ● Test suite failed to run
    TypeError: (0 , _api.getDefaultStartupMode) is not a function
      68 |     this.state = {
      69 |       language: getDefaultLanguage(),
    > 70 |       defaultMode: getDefaultStartupMode()
         |                    ^
      71 |     }
      72 | 
      73 |     this.onSaveSettings = this.onSaveSettings.bind(this)

**Comment by abahmed** [comment]
@sensokame see previous comment

**Comment by sensokame** [comment]
alright' I'll get it done asap

**Comment by sensokame** [comment]
> Could you update current screenshots too? @sensokame

am working on a windows machine, the screenshots (in the gif demo at least) are not, wouldn't that cause an issue?

**Comment by abahmed** [comment]
> am working on a windows machine, the screenshots (in the gif demo at least) are not, wouldn't that cause an issue?
@sensokame  It's OK, you can make a gif using online tools

File: app/utils/api.electron.js [hunk 1]
```diff
@@ -36,6 +37,33 @@ const getDefaultLanguage = function () {
...
+ * null.
+ */
+const getDefaultstartupNoteId = function () {
+  var mode = getDefaultStartupMode()
+  var noteId = 'general.' + mode
+  if (electronStore.has(noteId)) {
+    return electronStore.get(noteId)
+  }
+  if (mode === 'NEW') {
+    console.log('new')
```
Comment (abahmed) [inline app/utils/api.electron.js]: Remove logs

File: app/components/noteList/index.js [hunk 2]
```diff
@@ -41,11 +42,15 @@ const mapStateToProps = state => {
 }
 
 const mapDispatchToProps = dispatch => ({
-  setSelectedNoteID: (noteID) => dispatch(setSelectedNoteID(noteID)),
-  removeSelectedNote: () => dispatch(removeSelectedNote())
+  setSelectedNoteID: noteID => dispatch(setSelectedNoteID(noteID)),
+  removeSelectedNote: () => dispatch(removeSelectedNote()),
+  setCustomStartupNopte: noteID => setCustomNoteId(noteID)
```
Comment (abahmed) [inline app/components/noteList/index.js]: Fix typo : Nopte ==> Note

**Comment by sensokame** [comment]
@abahmed can you review it now?

**Comment by abahmed** [comment]
Thanks :-) @sensokame

**Pull request merged**